### DISCUSSION TITLE
Using dbeaver read-only with duckdb CLI (write mode) at the same time

### DISCUSSION THREAD

**Discussion by alanwilter** [duckdb/duckdb#11527] [Q&A] [answered]
Is that possible to use CLI `duckdb test.db` and at the same time having dbeaver accessing in read-only mode?
To setup dbeaver I followed [here](https://stackoverflow.com/questions/74411716/set-read-only-connection-to-duckdb-in-dbeaver)
I tried and I got a message about `Conflicting lock is held`. I read https://duckdb.org/docs/connect/concurrency but I actually didn't grasp the "concurrency" concept.
I can one run different programmes (like CLI duckdb, dbeaver, harlequin etc.) only if all in read-only mode?
If I want write mode so only one programme can be used at a time?

**Comment by Maxxen** [comment]
Yes

**Reply by alanwilter** [reply]
Sorry, yes to what? `If I want write mode so only one programme can be used at a time?`?

**Reply by Maxxen** [reply, marked as answer]
Yes

**Reply by Jimsparkle** [reply]
Yes
